src/ui/widgets/SlideControl/slideControl.tsx

```diff
@@ -1,115 +1,177 @@
-// Component to allow setting of values using a slider.
-// Displays value via an embedded progressbar widget
-
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import log from "loglevel";
 
-import classes from "./slideControl.module.css";
-
-import {
-  ProgressBarComponent,
-  ProgressBarProps
-} from "../ProgressBar/progressBar";
 import { writePv } from "../../hooks/useSubscription";
 import { Widget } from "../widget";
 import { PVInputComponent, PVWidgetPropType } from "../widgetProps";
-import { InferWidgetProps } from "../propTypes";
+import {
+  BoolPropOpt,
+  BorderPropOpt,
+  ColorPropOpt,
+  FloatPropOpt,
+  FontPropOpt,
+  InferWidgetProps,
+  IntPropOpt
+} from "../propTypes";
 import { registerWidget } from "../register";
 import { DType } from "../../../types/dtypes";
+import { Slider } from "@mui/material";
+import { diamondTheme } from "../../../diamondTheme";
+
+export const SliderControlProps = {
+  min: FloatPropOpt,
+  max: FloatPropOpt,
+  limitsFromPv: BoolPropOpt,
+  logScale: BoolPropOpt,
+  horizontal: BoolPropOpt,
+  showLabel: BoolPropOpt,
+  foregroundColor: ColorPropOpt,
+  backgroundColor: ColorPropOpt,
+  precision: IntPropOpt,
+  font: FontPropOpt,
+  border: BorderPropOpt,
+  height: FloatPropOpt,
+  width: FloatPropOpt,
+  enabled: BoolPropOpt,
+  transparent: BoolPropOpt,
+  levelHihi: FloatPropOpt,
+  levelHigh: FloatPropOpt,
+  levelLow: FloatPropOpt,
+  levelLolo: FloatPropOpt,
+  showHihi: BoolPropOpt,
+  showHigh: BoolPropOpt,
+  showLow: BoolPropOpt,
+  showLolo: BoolPropOpt,
+  increment: FloatPropOpt
```

Comment:
We're missing a `showScale` property here (which turns on or off the scale labels)